### PR TITLE
Updates Node pkg for netip.Addr

### PR DESCRIPTION
**Pull request by danehans**
Updates the node pkg to use `netip.Addr` instead of `net.IP`. The `AsSlice()` method is used in referent pkgs to minimize the size of this PR. As the referent pkgs are updated for `netip.Addr`, the `AsSlice()` method can be removed.

For: #24246

### CODE REVIEW

**Review by chancez — APPROVED** [review]

File: daemon/cmd/daemon.go [hunk 1]
```diff
@@ -326,15 +327,25 @@ func (d *Daemon) restoreCiliumHostIPs(ipv6 bool, fromK8s net.IP) {
...
 
-	restoredIP := node.RestoreHostIPs(ipv6, fromK8s, fromFS, cidrs)
-	if err := removeOldRouterState(ipv6, restoredIP); err != nil {
-		log.WithError(err).Warnf(
-			"Failed to remove old router IPs (restored IP: %s) from cilium_host. Manual intervention is required to remove all other old IPs.",
-			restoredIP,
-		)
+	k8sAddrs, k8sOk := ip.AddrFromIP(fromK8s)
+	if !k8sOk {
+		log.Warn("Failed to convert IP address %q", fromK8s)
```
Comment (YutaroHayakawa) [inline daemon/cmd/daemon.go:335]: Is it ok to ignore it and fall through to the next step? Same for the below one.

File: daemon/cmd/daemon.go [hunk 2]
```diff
@@ -558,11 +569,16 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 				oldNIDs = append(oldNIDs, nid)
 			} else if nid == identity.ReservedIdentityIngress && v.TunnelEndpoint.IsZero() {
 				oldIngressIPs = append(oldIngressIPs, k.IPNet())
-				ip := k.IPNet().IP
-				if ip.To4() != nil {
-					node.SetIngressIPv4(ip)
-				} else {
-					node.SetIngressIPv6(ip)
+				if addr, ok := netip.AddrFromSlice(k.IPNet().IP); ok {
```
Comment (YutaroHayakawa) [inline daemon/cmd/daemon.go:577]: Please don't use `AddrFromSlice` to convert net.IP to netip.Addr. Use ip.AddrFromIP instead.

Comment (YutaroHayakawa) [reply daemon/cmd/daemon.go:577]: We can see several other places using `AddrFromSlice`. Please fix them as well.

**Comment by danehans** [comment]
@YutaroHayakawa thanks for the review. I converted this PR back to draft while I work on fixing #24246 in smaller pkgs (see #25519 and #25521).

**Pull request closed without merging**